File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs(computers, visited, com):
    visited[com] = True
    for i in range(len(computers)):
        if computers[com][i] == 1 and not visited[i]:
            dfs(computers, visited, i)
            
def solution(n, computers):
    visited = [False] * n
    network = 0
    for com in range(n):
        if not visited[com]:
            dfs(computers, visited, com)
            network += 1
    return network

File: 프로그래머스/3/43162. 네트워크/네트워크.kt
class Solution {
    fun solution(n: Int, computers: Array<IntArray>): Int {
        val visited = BooleanArray(n){false}
        var networkCount = 0
        
        fun dfs(node:Int){
            visited[node] = true
            for(i in computers.indices){
                if(computers[node][i] == 1 && !visited[i]){
                    dfs(i)
                }
            }
        }
        
        for(i in 0 until n){
            if(!visited[i]){
                networkCount++
                dfs(i)
            }
        }
        return networkCount
    }
}